test/model/property_test.gd

```diff
@@ -146,6 +146,18 @@ func test_array_property_wrong_type() -> void:
 	new_property.load_data(property.save_data())
 	assert_that(new_property.get_default_value()).is_equal("")
 
+func test_array_property_custom_parsers() -> void:
```

Comment:
we probably also want a test that verifies that when the array is saved to json and then re-read, that the type is still there and not lost somehow. Could we extend one of the other existing tests (in entity backend) that deal with that sort of use case by adding a typed array there?